crates/rspack_core/src/compiler/compilation.rs

```diff
@@ -109,7 +109,7 @@ impl Compilation {
       module_graph,
       make_failed_dependencies: HashSet::default(),
       make_failed_module: HashSet::default(),
-      has_module_import_export_change: true,
+      has_module_import_export_change: false,
```

Comment:
I think setting to `true` will always make the result as expected, would this interfere the non-incremental build?

Reply:
https://github.com/web-infra-dev/rspack/blob/ed922e724e027484c565ccbedafaebfc15307291/crates/rspack_core/src/compiler/compilation.rs#L713-L727

The make phase will recalculate `has_module_import_export_change`. It is in a non-incremental build when the `origin_module_deps` is empty. The reason for this problem is that there is a call to `compilation.rebuildModule`, the previous `has_module_import_export_change` needs to be reused.

Reply:
https://github.com/web-infra-dev/rspack/blob/ed922e724e027484c565ccbedafaebfc15307291/crates/rspack_core/src/compiler/hmr.rs#L144-L197

I could move this setting to incremental checks and it would solve the problem too, but I just think false as the default is more intuitive

Reply:
Yes. I have modified this into incremental initialization